playbooks/common/openshift-master/config.yml

```diff
@@ -337,12 +361,54 @@
     with_items: openshift_master_named_certificates
     when: named_certs_specified | bool
 
-- name: Configure master instances
-  hosts: oo_masters_to_config
+- name: Configure first master instance
```

Comment:
> So how can we just use one play here? It looked to me like I must (a) calculate openshift_version based on all other inputs for first master, (b) install first master so it has a openshift.common.version fact, and then (c) install all other masters. I can't think of a way to do this without separation.

You could do this by using set_fact to conditionally set it based on whether the host is the "first master". The template in the set_fact task will be evaluated at run time, and as long as the play is still serialized we would know that we have already run on the first master.

Also, since the act of running openshift_facts updates the local facts for the host, by the time the subsequent masters are run, the fact variable would already be set.

> I also just noticed the subsequent masters are not properly protecting their installed versions, i.e. if first master somehow got a newer version, a re-run of config playbook would upgrade the subsequent masters. Adding that to TODO as well.

I'm not completely convinced they should, either. if the config playbooks are intended to keep the cluster configured and in sync, the version from the first master should be the starting point, I believe.